modules/local/align_bwa/main.nf

```diff
@@ -25,52 +25,101 @@ process INDEX_FASTA {
     """
 }
 
-process ALIGNMENT_BAM {
+process FEATURED_ALIGNMENT {
 
     /*
     This module aligns reads to the reference using specified arguments and produces output in the form of FASTQ.GZ, BAM and BAM.BAI files.
+    It also runs:
+     - depth generation using jgi_summarize_bam_contig_depths if get_depth=true
+     - concoct coverage table if get_concoct_table=true
     */
 
     label 'process_medium'
 
     tag "${meta.id} align to ${ref_fasta}"
 
-    container 'quay.io/microbiome-informatics/bwamem2:2.2.1'
+    container 'quay.io/microbiome-informatics/bwa_metabat_concoct:2.2.1_2.16_1.1.0'
 
     input:
     tuple val(meta), path(reads), path(ref_fasta), path(ref_fasta_index)
+    val(get_depth)
+    val(get_concoct_table)
+    val(bed)
 
     output:
-    tuple val(meta), path(ref_fasta), path("output/${meta.id}_sorted.bam"), path("output/${meta.id}_sorted.bam.bai"), emit: bam
+    // tuple val(meta), path(ref_fasta), path("output/${meta.id}_sorted.bam"), path("output/${meta.id}_sorted.bam.bai"), emit: bam
```

Comment:
Remove the commented line and align emits. Move `path "versions.yml" ` as the last output just for readability.